src/vs/workbench/contrib/chat/browser/chatContentParts/toolInvocationParts/chatToolProgressPart.ts

```diff
@@ -32,15 +37,21 @@ export class ChatToolProgressSubPart extends BaseChatToolInvocationSubPart {
 
 	private createProgressPart(): HTMLElement {
 		if (this.toolInvocation.isComplete && this.toolIsConfirmed && this.toolInvocation.pastTenseMessage) {
-			const part = this.renderProgressContent(this.toolInvocation.pastTenseMessage);
+			const key = this.getAnnouncementKey('complete');
+			const completionContent = this.toolInvocation.pastTenseMessage ?? this.toolInvocation.invocationMessage;
+			const shouldAnnounce = this.toolInvocation.kind === 'toolInvocation' && this.hasMeaningfulContent(this.toolInvocation.pastTenseMessage) ? this.computeShouldAnnounce(key) : false;
+			const part = this.renderProgressContent(completionContent, shouldAnnounce);
 			this._register(part);
 			return part.domNode;
 		} else {
 			const container = document.createElement('div');
 			const progressObservable = this.toolInvocation.kind === 'toolInvocation' ? this.toolInvocation.progress : undefined;
 			this._register(autorun(reader => {
 				const progress = progressObservable?.read(reader);
-				const part = reader.store.add(this.renderProgressContent(progress?.message || this.toolInvocation.invocationMessage));
+				const key = this.getAnnouncementKey('progress');
+				const progressContent = progress?.message ?? this.toolInvocation.invocationMessage;
+				const shouldAnnounce = this.toolInvocation.kind === 'toolInvocation' && this.hasMeaningfulContent(progress?.message) ? this.computeShouldAnnounce(key) : false;
```

Comment:
Just from looking at the code, it looks like if the chat is not being rendered while the response loads (sidebar collapsed, or chat is scrolled), then all of these would be read simultaneously as soon as they are rendered.

If that's an issue then you'd want to find another way to decide whether to read a particular progress item. `kind === 'toolInvocation'` probably works fine today to decide not read an item from a reloaded session, but I don't think it's safe, there's no real guarantee that this type is only used for new tool calls.

Reply:
I tried what you suggested and not sure if I'm not hitting the case you're concerned about bc i'm forced to sign in again or if that's not an issue. Can you try pls?

Reply:
Yeah I tried those and did not have an issue